pkg/planner/core/plan_cost_ver2.go

```diff
@@ -160,6 +161,13 @@ func getPlanCostVer24PhysicalTableScan(pp base.PhysicalPlan, taskType property.T
 	} else { // TiFlash
 		columns = p.Schema().Columns
 	}
+	// _tidb_commit_ts is not a real extra column stored in the disk, and it should not bring extra cost, so we exclude
+	// it from the cost here.
+	for i, col := range columns {
+		if col.ID == model.ExtraCommitTSID {
+			columns = slices.Delete(slices.Clone(columns), i, i+1)
+		}
+	}
```

Comment:
ditto
it's better to handle it in `getAvgRowSize`?

Reply:
`getAvgRowSize` is used in too many other places, like costing for `Sort` and `Aggregation`. I think we should only ignore its cost of scanning. So I tend to keep it here.